integration/aggregate_compat_test.go

```diff
@@ -656,9 +648,10 @@ func TestAggregateCompatGroupDocDotNotation(t *testing.T) {
 
 	testCases := map[string]aggregateStagesCompatTestCase{
 		"DocDotNotation": {
-			pipeline: bson.A{bson.D{{"$group", bson.D{
-				{"_id", "$v.foo"},
-			}}}},
+			pipeline: bson.A{
+				bson.D{{"$sort", bson.D{{"_id", 1}}}},
```

Comment:
If our providers don't have _id with arrays (and they should not) and if our sorting is stable for scalar values, then sorting by _id before grouping should do the trick

Reply:
Okay I see what you mean, I finally understood. The result of aggregation should in the order it was given so as long as input is sorted uniquely, we will always get deterministic result. Thanks for this!